sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/aad/webapp/AADWebAppConfiguration.java

```diff
@@ -43,6 +44,7 @@
 @Configuration
 @ConditionalOnMissingClass({ "org.springframework.security.oauth2.server.resource.BearerTokenAuthenticationToken" })
 @ConditionalOnClass(ClientRegistrationRepository.class)
+@ConditionalOnProperty("azure.activedirectory.client-id")
```

Comment:
We'd better add a unit test for this case.

Reply:
I also find the same problem in obo auto-configuration, please hold the release for my PR.